service/framework.go

```diff
@@ -252,7 +253,22 @@ func newCustomObjectFramework(config Config) (*framework.Framework, error) {
 	}
 
 	initCtxFunc := func(ctx context.Context, obj interface{}) (context.Context, error) {
-		ctx = messagecontext.NewContext(ctx, messagecontext.NewMessage())
+		{
+			message := messagecontext.NewMessage()
+			ctx = messagecontext.NewContext(ctx, message)
+		}
+
+		{
+			container := loggercontext.NewContainer()
+
+			customObject, err := key.ToCustomObject(obj)
+			if err != nil {
+				return nil, microerror.Mask(err)
+			}
+			container.KeyVals["cluster"] = key.ClusterID(customObject)
+
+			ctx = loggercontext.NewContext(ctx, container)
+		}
```

Comment:
Here is the context prepared and the cluster ID is set based on the custom object. Every log issuance using the context can now omit additional garbage and the logs become more consistent.